modules/core/02-client/keeper/client.go

```diff
@@ -36,17 +33,16 @@ func (k Keeper) CreateClient(
 		return "", err
 	}
 
+	k.SetClientState(ctx, clientID, clientState)
 	k.SetClientConsensusState(ctx, clientID, clientState.GetLatestHeight(), consensusState)
 
 	k.Logger(ctx).Info("client created at height", "client-id", clientID, "height", clientState.GetLatestHeight().String())
 
-	defer func() {
-		telemetry.IncrCounterWithLabels(
-			[]string{"ibc", "client", "create"},
-			1,
-			[]metrics.Label{telemetry.NewLabel(types.LabelClientType, clientState.ClientType())},
-		)
-	}()
+	defer telemetry.IncrCounterWithLabels(
```

Comment:
this was not mentioned in the audit however we are wrapping these calls with an anonymous function which is not required.

Reply:
Ah, nice catch. I think then this improvement could be applied to other places of the codebase as well.